src/Access/Common/AuthenticationData.cpp

```diff
@@ -85,6 +95,40 @@ AuthenticationData::Digest AuthenticationData::Util::encodeSHA256(std::string_vi
 #endif
 }
 
+AuthenticationData::Digest AuthenticationData::Util::encodeBcrypt(std::string_view text [[maybe_unused]])
+{
+#if USE_BCRYPT
+    char salt[BCRYPT_HASHSIZE];
+    Digest hash;
+    hash.resize(64);
+    int ret = bcrypt_gensalt(12, salt);
+    if (ret != 0)
+        throw DB::Exception(DB::ErrorCodes::LOGICAL_ERROR, "BCrypt library failed: bcrypt_gensalt returned {}", ret);
+    ret = bcrypt_hashpw(text.data(), salt, reinterpret_cast<char *>(hash.data()));
```

Comment:
Bcrypt password has a limit of 72 characters, anything after 72 characters is ignored - perhaps we can check for that and throw an error?